### DISCUSSION TITLE
SHAP Value with one-hot encoding by LightGBM

### DISCUSSION THREAD

**Discussion by hugolam1** [shap/shap#3381] [Q&A]
I am struggling with 2 main focuses in my modelling work: 

1. Should I use one-hot encoding in LightGBM to show all the category's importance, would it cause multicollinearity? 

3. How can I show the SHAP value importance if I used one-hot encoding in my dataset and how do I know if the categorical feature has a positive/negative impact? 

I saw some of the posts wanting to group back the categorical feature after the encoding, but I would like to focus on all categories in the categorical features. 

For example, the variable: "industry" - the categories after one-hot encoding are: industry_finacnce, industry_tech and industry_manufacturing. 

My goal is to show these 3 categories in the shap value importance plot. 

Does anyone have any idea about these? Appreciate your help in advance!!

**Comment by gabrown** [comment]
@hugolam1  Did you solve this issue? I want to do something very similar
